java/code/src/com/redhat/rhn/common/hibernate/HibernateFactory.java

```diff
@@ -780,11 +784,33 @@ protected static void executeCallableMode(String name, String mode, Map params)
         return findByIds(ids, queryName, idsParameterName, new HashMap<String, Object>());
     }
 
+    /**
+     * Executes an 'update' query to the database given a list of parameters.
+     * The query will be execute in batches of LIST_BATCH_MAX_SIZE parameters each.
+     * @param <E> the type of the list parameters
+     * @param list the list of parameters to search for
+     * @param queryName the name of the query to be executed
+     * @param parameterName the name of the parameter to match the parameters in the list
+     * @return the count of affected rows
+     */
+    @SuppressWarnings("unchecked")
+    protected static <E> int udpateByIds(List<E> list, String queryName, String parameterName,
+            Map<String, Object> parameters) {
+        Query<Integer> query = HibernateFactory.getSession()
+                .getNamedQuery(queryName);
+
+        parameters.entrySet().stream().forEach(entry -> {
+            query.setParameter(entry.getKey(), entry.getValue());
+        });
+
+        return splitAndExecuteQuery(list, parameterName, query, query::executeUpdate, 0, Integer::sum);
+    }
+
     /**
      * Executes a 'lookup' query to retrieve data from the database given a list of ids.
-     * The query will be execute in batchs of 1000 ids each.
+     * The query will be execute in batchs of LIST_BATCH_MAX_SIZE ids each.
```

Comment:
typo : batchs -> batches

Reply:
it's not a typo, https://simple.wiktionary.org/wiki/batches :-)